CMake/FindAOFlagger.cmake

```diff
@@ -0,0 +1,105 @@
+# This is a FindPackage script for AOFlagger.
+#
+# It sets the following variables:
+# - AOFLAGGER_INCLUDE_DIR
+# - AOFLAGGER_LIB
+# - AOFLAGGER_VERSION[_MAJOR/_MINOR]
+# - AOFLAGGER_FOUND
+# - AOFLAGGER_ROOT_DIR (also used as hint)
+# The variable AOFLAGGER_ROOT_DIR can be set to select a specific version of aoflagger.
+#
+# A few dependencies of AOFlagger are not handled by this script, because they are complex and
+# often already included by the caller. These are: casacore, Boost, Python
+# These need to be manually configured by the caller.
+
+cmake_minimum_required(VERSION 3.0)
+
+set(AOFLAGGER_FOUND FALSE)
+
+find_library(AOFLAGGER_LIB aoflagger
+  HINTS ${AOFLAGGER_ROOT_DIR} PATH_SUFFIXES lib)
+find_path(AOFLAGGER_INCLUDE_DIR NAMES aoflagger.h
+  HINTS ${AOFLAGGER_ROOT_DIR} PATH_SUFFIXES include)
+execute_process(COMMAND ${AOFLAGGER_INCLUDE_DIR}/../bin/aoflagger --version OUTPUT_VARIABLE _aoflagger_result)
+if(_aoflagger_result)
+  get_filename_component(AOFLAGGER_ROOT_DIR ${AOFLAGGER_INCLUDE_DIR} DIRECTORY)
+  # AOFlagger --version returns something like
+  # "AOFlagger 3.0-alpha", hence split space and dash:
+  string(REPLACE "-" " " _aoflagger_result ${_aoflagger_result})
+  string(REPLACE " " ";" _aoflagger_reslist ${_aoflagger_result})
+  list(GET _aoflagger_reslist 1 AOFLAGGER_VERSION)
+  string(REPLACE "." ";" _aoflagger_version_list ${AOFLAGGER_VERSION})
+  list(GET _aoflagger_version_list 0 AOFLAGGER_VERSION_MAJOR)
+  list(GET _aoflagger_version_list 1 AOFLAGGER_VERSION_MINOR)
+  message(STATUS "Found AOFlagger version ${AOFLAGGER_VERSION_MAJOR}.${AOFLAGGER_VERSION_MINOR}: ${AOFLAGGER_LIB}")
+else()
+  message(STATUS "  Executing process aoflagger failed -- AOFlagger set to NOT FOUND.")
+endif()
+
+if(AOFLAGGER_LIB AND AOFLAGGER_INCLUDE_DIR AND AOFLAGGER_VERSION)
+  set(_dependencies_met TRUE)
+  #
+  # Dependencies of AOFlagger
+  #
+  find_package(LibXml2)
+  list(APPEND AOFLAGGER_LIB ${LibXml2_LIB})
+  if(NOT LibXml2_FOUND)
+    message(STATUS "AOFlagger dependency LibXml2 not found.")
+    set(_dependencies_met FALSE)
+  endif()
+  
+  find_package(PNG)
+  list(APPEND AOFLAGGER_LIB ${PNG_LIBRARIES})
+  if(NOT PNG_FOUND)
+    message(STATUS "AOFlagger dependency PNG not found.")
+    set(_dependencies_met FALSE)
+  endif()
+  
+  find_library(FFTW3_LIB fftw3 REQUIRED)
+  list(APPEND AOFLAGGER_LIB ${FFTW3_LIB})
+  if(NOT FFTW3_LIB)
+    message(STATUS "AOFlagger dependency FFTW3 not found.")
+    set(_dependencies_met FALSE)
+  endif()
+  
+  find_package(CFITSIO)
+  if(CFITSIO_FOUND)
+    list(APPEND AOFLAGGER_LIB ${CFITSIO_LIBRARY})
+    list(APPEND AOFLAGGER_INCLUDE_DIR ${CFITSIO_INCLUDE_DIR})
+  elseif()
+    message(STATUS "AOFlagger dependency CFITSIO not found.")
+    set(_dependencies_met FALSE)
+  endif()
+
+  find_package(PkgConfig)
+  pkg_check_modules(GTKMM gtkmm-3.0>=3.0.0)
+  pkg_check_modules(SIGCXX sigc++-2.0)
```

Comment:
Why do you check sigc++ ? It's not used in the remainder of this file.

I see it was in the original CMakeLists.txt, but there it's also unclear why sigc++ is there.

Reply:
Thanks, good point. AOFlagger searches for it too and links with it when it is found. It's also part of GTKMM I believe, maybe requiring it changed over time. I've removed it now.